force-app/main/default/classes/OSF_UpdateAccountsBatch.cls

```diff
@@ -0,0 +1,91 @@
+/**********************************************************************
+ * @Name         : OSF_UpdateAccountsBatch
+ * @Description  : A batch class that modify Accounts Weather data
+ * @Created By   : Marta Demeny
+ * @Created Date : 13 jun. 2022
+ * @Modification Log: 
+ ********************************************************************
+ * Version		Developer		Date			Description
+ *___________________________________________________________________
+ *
+ *********************************************************************/
+
+global with sharing class OSF_UpdateAccountsBatch implements Database.Batchable<sObject>, Database.Stateful, Database.AllowsCallouts {
+    // instance member to retain state across transactions
+    public Integer recordsProcessed = 0;
+
+    private static final String WEATHER_IMG_URL = 'https://openweathermap.org/img/wn/';
```

Comment:
I deleted this variable to don't read again from custom setting. The weather info data are return using the OSF_WeatherReader.getWeatherInfo method instead of OSF_WeatherReader.executeAPICall